daemon/logger/omslogs/omslogs.go

```diff
@@ -0,0 +1,350 @@
+package omslogs
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"os"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/docker/docker/daemon/logger"
+	client "github.com/docker/docker/daemon/logger/omslogs/omsclient"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	name = "omslogs"
+
+	// Command line options
+	optDomain      = "omslogs-domain"
+	optWorkspaceID = "omslogs-workspaceID"
+	optSharedKey   = "omslogs-sharedKey"
+	optTimeout     = "omslogs-timeout"
+
+	// Environment variables
+	envDomain                = "OMS_LOGGING_DOMAIN"
+	envTimeout               = "OMS_LOGGING_DRIVER_TIMEOUT"
+	envPostMessagesFrequency = "OMS_LOGGING_DRIVER_POST_MESSAGES_FREQUENCY"
+	envPostMessagesBatchSize = "OMS_LOGGING_DRIVER_POST_MESSAGES_BATCH_SIZE"
+	envBufferSize            = "OMS_LOGGING_DRIVER_BUFFER_SIZE"
+	envStreamChannelSize     = "OMS_LOGGING_DRIVER_CHANNEL_SIZE"
+
+	// Default option values
+	defaultTimeout               = time.Duration(5 * time.Second)
+	defaultPostMessagesFrequency = time.Duration(5 * time.Second)
```

Comment:
Looks like these should just be `5 * time.Second` (as `time.Second` is already a duration?)